pkg/endpoint/endpoint.go

```diff
@@ -34,28 +35,60 @@ type Config struct {
 
 	tls.Config
 }
-
 type ETCDConfig struct {
 	Endpoints   []string
 	TLSConfig   tls.Config
 	LeaderElect bool
 }
 
 func Listen(ctx context.Context, config Config) (ETCDConfig, error) {
-	driver, dsn := ParseStorageEndpoint(config.Endpoint)
+	driver, dsn, err := ParseStorageEndpoint(config.Endpoint)
 	if driver == ETCDBackend {
 		return ETCDConfig{
 			Endpoints:   strings.Split(config.Endpoint, ","),
 			TLSConfig:   config.Config,
 			LeaderElect: true,
 		}, nil
 	}
+	if err != nil {
+		return ETCDConfig{}, errors.Wrap(err, "parsing storage point")
+	}
+
+	type Response struct {
```

Comment:
Sorry, I don't fully understand the context, but it really doesn't seem like the right place to actually be testing the connection.  What is the issue you are trying to fix that requires testing the connection?